src/mibase.ts

```diff
@@ -776,7 +776,80 @@ export class MI2DebugSession extends DebugSession {
 			this.sourceFileMap = new SourceFileMap(configMap, fallbackGDB);
 		}
 	}
+	public async createIntegratedTerminalLinux(args:DebugProtocol.RunInTerminalRequestArguments) {
+		const mkdirAsync = fs.promises.mkdir;
+		const mkdtempAsync = async (tempDir: string, prefix: string): Promise<string> => {
+			const name = `${prefix}-${Date.now()}-${Math.floor(Math.random() * 1e9)}`;
+			const newDirPath = `${tempDir}/${name}`;
 
+			try {
+				await mkdirAsync(newDirPath, { recursive: true });
+				return newDirPath;
+			} catch (err) {
+				throw new Error(`Error creating temp directory: ${err.message}`);
+			}
+		};
+		const ttyTmpDir = await mkdtempAsync(os.tmpdir(), 'debug');
+
+		(async () => {
+			try {
+				fs.writeFileSync(
+					`${ttyTmpDir}/get-tty`,
+					`#!/usr/bin/env sh
+					# reset terminal to clear any previous states
+					reset
+					echo "The input and output of program will be here."
+					echo "Warning of set controlling terminal fail can be ignored."
+					tty > ${ttyTmpDir}/ttynameTmp
+					mv ${ttyTmpDir}/ttynameTmp ${ttyTmpDir}/ttyname
```

Comment:
Just fo0r understanding: Is the first command slow and the `mv` is used to ensure it is finished before it is available under the name?

Reply:
the first command (tty > ${ttyTmpDir}/ttynameTmp) is not inherently slow. The use of mv here is not primarily about addressing slowness but rather about ensuring atomicity and consistency